src/encoding.rs

```diff
@@ -536,6 +537,12 @@ where
     }
 }
 
+impl EncodeLabelValue for bool {
+    fn encode(&self, encoder: &mut LabelValueEncoder) -> Result<(), std::fmt::Error> {
+        encoder.write_str(if *self { "true" } else { "false" })
+    }
+}
```

Comment:
for the record, i would support this addition, i think it's a pleasant and natural addition to the library.

that said, the reordering of `impl` blocks here is probably best left for a distinct pull request, as it introduces a lot of undue noise for reviewers to inspect.